### ISSUE TITLE
unable to run gizmo; panic: oto: NewContext can be called only once

### ISSUE THREAD

**Issue by mewmew** [Lallassu/gizmo#3]
Hi!

I was very excited to read about Gizmo when you posted about it on Reddit a while back! The graphics style reminds me of [Liero](https://en.wikipedia.org/wiki/Liero), a game my friends and I loved to play when we were younger.

So, when I saw you released the code for the game, I felt very excited to take it out for a spin :)

However, after the initial `go get -u github.com/Lallassu/gizmo/...`, running `gizmo` (from within `$GOPATH/src/github.com/Lallassu/gizmo` results in the following error:

```
u@x1 ~/g/s/g/L/gizmo> gizmo
[23:40:43.204][main.go:31][main.run][DEBUG] W: Rect(0, 0, 800, 600)
[23:40:43.242][sound.go:23][main.(*sound).create][DEBUG] Loading sound jump.wav
[23:40:43.242][sound.go:23][main.(*sound).create][DEBUG] Loading sound shot.mp3
[23:40:43.314][sound.go:23][main.(*sound).create][DEBUG] Loading sound shot2.mp3
panic: oto: NewContext can be called only once

goroutine 6 [running]:
github.com/hajimehoshi/oto.NewContext(0xbb80, 0x2, 0x2, 0x31fc, 0x0, 0x0, 0x0)
	/home/u/goget/src/github.com/hajimehoshi/oto/context.go:60 +0x311
github.com/hajimehoshi/oto.NewPlayer(0xbb80, 0x2, 0x2, 0x31fc, 0xc0004121d8, 0xc000514000, 0xc000475ff0)
	/home/u/goget/src/github.com/hajimehoshi/oto/context.go:89 +0x4d
github.com/faiface/beep/speaker.Init(0xbb80, 0xc7f, 0x0, 0x0)
	/home/u/goget/src/github.com/faiface/beep/speaker/speaker.go:42 +0x154
main.(*sound).create(0x7e2aa8)
	/home/u/goget/src/github.com/Lallassu/gizmo/sound.go:32 +0x56a
main.setup()
	/home/u/goget/src/github.com/Lallassu/gizmo/main.go:66 +0x12f
main.run()
	/home/u/goget/src/github.com/Lallassu/gizmo/main.go:43 +0x313
github.com/faiface/mainthread.Run.func1(0x6622f8, 0xc00001a120)
	/home/u/goget/src/github.com/faiface/mainthread/mainthread.go:37 +0x27
created by github.com/faiface/mainthread.Run
	/home/u/goget/src/github.com/faiface/mainthread/mainthread.go:36 +0xb1
```

Cheers,
Robin

**Comment by Lallassu** [comment]
Hi, this issue is already posted and seems to be a bug in Windows. I've only developed and tested on OSX. I will mark this a duplicate.

A workaround would be to remove the loading of the sounds.

**Comment by Lallassu** [comment]
duplicate of #2

**Comment by mewmew** [comment]
> Hi, this issue is already posted and seems to be a bug in Windows

Sorry for the duplicate bug report.

I'm running on Linux, though, so it seems the issue is more pervasive.

**Comment by Lallassu** [comment]
Ok good to know. Though it seems to be an issue with the sound loader so I will check that out later. Sounds are not activated yet so it can be removed at the moment.

**Comment by faiface** [comment]
Hi, this also should be fixed with the recent Beep update [here](https://github.com/faiface/beep/commit/fc2da0b5e081c0c574efd339344ad9f1bf649cba)

**Comment by mewmew** [comment]
I get further now (with the latest version of Beep), could start the game menu and create a new game :)

![foo](https://user-images.githubusercontent.com/1414531/53640447-9035eb80-3c6f-11e9-969d-5dbe454f433d.png)

However, when I run the game a second and third time, I got the following error:

```
u@x1 ~/g/s/g/L/gizmo> gizmo
[22:12:34.022][main.go:31][main.run][DEBUG] W: Rect(0, 0, 800, 600)
[22:12:34.042][sound.go:23][main.(*sound).create][DEBUG] Loading sound jump.wav
[22:12:34.042][sound.go:23][main.(*sound).create][DEBUG] Loading sound shot.mp3
[22:12:34.117][sound.go:23][main.(*sound).create][DEBUG] Loading sound shot2.mp3
fatal error: unexpected signal during runtime execution
[signal SIGSEGV: segmentation violation code=0x1 addr=0x3c pc=0x7fe327797bfa]

runtime stack:
runtime.throw(0x65ec43, 0x2a)
	/home/u/go1.12/src/runtime/panic.go:617 +0x72
runtime.sigpanic()
	/home/u/go1.12/src/runtime/signal_unix.go:374 +0x4a9

goroutine 4 [syscall]:
runtime.cgocall(0x5b56a0, 0xc000044d88, 0xc000322000)
	/home/u/go1.12/src/runtime/cgocall.go:128 +0x5b fp=0xc000044d58 sp=0xc000044d20 pc=0x427c9b
github.com/hajimehoshi/oto._Cfunc_snd_pcm_writei(0x7fe300049170, 0xc000322000, 0x42a, 0x0)
	_cgo_gotypes.go:160 +0x4e fp=0xc000044d88 sp=0xc000044d58 pc=0x55b4ce
github.com/hajimehoshi/oto.(*driver).TryWrite.func1(0xc0004162c0, 0xc00051c000)
	/home/u/goget/src/github.com/hajimehoshi/oto/driver_linux.go:140 +0x123 fp=0xc000044de0 sp=0xc000044d88 pc=0x55c2d3
github.com/hajimehoshi/oto.(*driver).TryWrite(0xc0004162c0, 0xc00051c0a0, 0x60, 0x8000, 0x100, 0x0, 0x0)
	/home/u/goget/src/github.com/hajimehoshi/oto/driver_linux.go:140 +0x1e0 fp=0xc000044e78 sp=0xc000044de0 pc=0x55bc80
github.com/hajimehoshi/oto.(*driverWriter).Write(0xc00041a020, 0xc00051c000, 0x100, 0x8000, 0x0, 0x0, 0x0)
	/home/u/goget/src/github.com/hajimehoshi/oto/context.go:136 +0xf4 fp=0xc000044ef0 sp=0xc000044e78 pc=0x55a984
io.copyBuffer(0x68d440, 0xc00041a020, 0x68d460, 0xc000416300, 0xc00051c000, 0x8000, 0x8000, 0x0, 0x0, 0x0)
	/home/u/go1.12/src/io/io.go:404 +0x1fb fp=0xc000044f68 sp=0xc000044ef0 pc=0x4b239b
io.Copy(...)
	/home/u/go1.12/src/io/io.go:364
github.com/hajimehoshi/oto.NewContext.func1(0xc00041a040)
	/home/u/goget/src/github.com/hajimehoshi/oto/context.go:79 +0x6a fp=0xc000044fd8 sp=0xc000044f68 pc=0x55bf7a
runtime.goexit()
	/home/u/go1.12/src/runtime/asm_amd64.s:1337 +0x1 fp=0xc000044fe0 sp=0xc000044fd8 pc=0x47ae01
created by github.com/hajimehoshi/oto.NewContext
	/home/u/goget/src/github.com/hajimehoshi/oto/context.go:78 +0x229

goroutine 1 [select, locked to thread]:
github.com/faiface/mainthread.Run(0x6622f8)
	/home/u/goget/src/github.com/faiface/mainthread/mainthread.go:42 +0x13d
github.com/faiface/pixel/pixelgl.Run(0x6622f8)
	/home/u/goget/src/github.com/faiface/pixel/pixelgl/run.go:32 +0x57
main.main()
	/home/u/goget/src/github.com/Lallassu/gizmo/main.go:14 +0x2d

goroutine 20 [syscall]:
os/signal.signal_recv(0x0)
	/home/u/go1.12/src/runtime/sigqueue.go:139 +0x9c
os/signal.loop()
	/home/u/go1.12/src/os/signal/signal_unix.go:23 +0x22
created by os/signal.init.0
	/home/u/go1.12/src/os/signal/signal_unix.go:29 +0x41

goroutine 21 [chan receive]:
github.com/hajimehoshi/oto.(*Context).Close(0xc00041a040, 0x0, 0x0)
	/home/u/goget/src/github.com/hajimehoshi/oto/context.go:116 +0xd2
github.com/faiface/beep/speaker.Init(0xbb80, 0xc7f, 0x0, 0x0)
	/home/u/goget/src/github.com/faiface/beep/speaker/speaker.go:34 +0x2e8
main.(*sound).create(0x7e2ab0)
	/home/u/goget/src/github.com/Lallassu/gizmo/sound.go:32 +0x56a
main.setup()
	/home/u/goget/src/github.com/Lallassu/gizmo/main.go:66 +0x12f
main.run()
	/home/u/goget/src/github.com/Lallassu/gizmo/main.go:43 +0x313
github.com/faiface/mainthread.Run.func1(0x6622f8, 0xc0000760c0)
	/home/u/goget/src/github.com/faiface/mainthread/mainthread.go:37 +0x27
created by github.com/faiface/mainthread.Run
	/home/u/goget/src/github.com/faiface/mainthread/mainthread.go:36 +0xb1
```

**Comment by faiface** [comment]
Well, that's very interesting. I'll try the game on my system and let you know.
